pkg/cmd/step/verify/step_verify_environments.go

```diff
@@ -312,20 +252,119 @@ func (o *StepVerifyEnvironmentsOptions) createEnvGitRepository(name string, requ
 			ServerKind:               gitKind,
 			Username:                 userAuth.Username,
 			ApiToken:                 userAuth.Password,
-			Owner:                    gitInfo.Organisation,
-			RepoName:                 gitInfo.Name,
-			Public:                   publicRepo,
+			Owner:                    envGitInfo.Organisation,
+			RepoName:                 envGitInfo.Name,
+			Public:                   public,
 			IgnoreExistingRepository: true,
 		}
 
-		_, _, err = kube.DoCreateEnvironmentGitRepo(batchMode, authConfigSvc, environment, forkGitURL, envDir, gitRepoOptions, helmValues, prefix, o.Git(), o.ResolveChartMuseumURL, o.In, o.Out, o.Err)
+		_, _, err = kube.DoCreateEnvironmentGitRepo(batchMode, authConfigSvc, environment, forkGitURL, envDir, gitRepoOptions, helmValues, prefix, gitter, o.ResolveChartMuseumURL, o.In, o.Out, o.Err)
 		if err != nil {
 			return errors.Wrapf(err, "failed to create git repository for gitURL %s", gitURL)
 		}
 	}
 	return nil
 }
 
+func (o *StepVerifyEnvironmentsOptions) createDevEnvironmentRepository(envGitInfo *gits.GitRepository, public bool, provider gits.GitProvider, gitter gits.Gitter, requirements *config.RequirementsConfig) error {
+	fromGitURL, fromBaseRef, err := o.readEnvironment()
+	if err != nil {
+		return err
+	}
+
+	dir, err := filepath.Abs(o.Dir)
+	if err != nil {
+		return errors.Wrapf(err, "resolving %s to absolute path", o.Dir)
+	}
+
+	environmentRepo, err := o.createDevEnvironmentRemote(envGitInfo, dir, fromGitURL, fromBaseRef, !public, requirements, provider, gitter)
```

Comment:
@pmuir one thing I don't understand is why we create the remote and push the original repo data to it, but ony in the `jx boot` case add a commit with the modifications? 

I also find it odd that we push something to the remote, then fork and pull, add a commit and then push again. Could we not do everything locally on then do a single push? 

Apart from this, afaict the refactored code does not what the original code did, except that the changes to config are pushed to master instead of creating a pull request. 

Still cannot shake off the feeling that we only should push once.